app/controllers/simplified-account/settings/team-members/invite/invite.controller.js

```diff
@@ -2,21 +2,64 @@ const { response } = require('@utils/response')
 const { getAvailableRolesForService } = require('@utils/roles')
 const formatSimplifiedAccountPathsFor = require('@utils/simplified-account/format/format-simplified-account-paths-for')
 const paths = require('@root/paths')
+const { body, validationResult } = require('express-validator')
+const formatValidationErrors = require('@utils/simplified-account/format/format-validation-errors')
+const userService = require('@services/user.service')
+const lodash = require('lodash')
 
 async function get (req, res) {
   const serviceId = req.service.externalId
   const accountType = req.account.type
-  const serviceHasAgentInitiatedMotoEnabled = req.service.agentInitiatedMotoEnabled ?? false
-  const availableRoles = getAvailableRolesForService(serviceHasAgentInitiatedMotoEnabled)
   response(req, res, 'simplified-account/settings/team-members/invite',
     {
-      availableRoles,
+      availableRoles: getAvailableRolesForService(req.service.agentInitiatedMotoEnabled ?? false),
       backLink: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.teamMembers.index, serviceId, accountType)
     })
 }
 
 async function post (req, res, next) {
-  // TODO
+  const externalServiceId = req.service.externalId
+  const accountType = req.account.type
+  const adminUserExternalId = req.user.externalId
+  const invitedUserEmail = req.body.invitedUserEmail
+  const invitedUserRole = req.body.invitedUserRole
+  const teamMembersIndexPath = formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.teamMembers.index, externalServiceId, accountType)
+
+  const validations = [
+    body('invitedUserRole')
+      .not().isEmpty().withMessage('Select a permission level'),
+    body('invitedUserEmail').isEmail().withMessage('Enter a valid email address')
```

Comment:
I think it's worth adding a length constraint as well, you can just do `isLength({ max: <length> })` - the database field is a `VARCHAR(255)` so it'll fail on emails longer than that anyway, but we may as well catch it earlier.

You can also use `bail()` like
```
body('invitedUserEmail')
  .isLength({ ... }).withMessage('Enter an email address').bail()
  .isEmail().withMessage('Enter a valid email address')
```

which will mean it only returns the first error if there is no email entered, rather than both.

Reply:
I've looked into this and the [isEmail validation](https://github.com/validatorjs/validator.js/blob/86911d817fbcd177c295b13d7ca0fdff87341191/src/lib/isEmail.js#L29) already checks for a default max length 254 which is now standard (as well as a maximum for the user part and domain part), so I don't think we need to validate length separately.

Reply:
ah very nice, no need to duplicate that then. good find